crates/dbsp/src/trace/layers/erased/mod.rs

```diff
@@ -282,9 +289,12 @@ impl ErasedLayer {
             diffs.set_len(self.diffs.len());
         }
 
+        // TODO: We can eliminate elements from `0..lower_bound` when creating the
```

Comment:
Nope, in general we can't, as this will screw up the offsets in the parent layer.